File: components/credentials/src/main/java/org/cloudfoundry/credhub/audit/entities/DeleteCredential.java
package org.cloudfoundry.credhub.audit.entities;

import org.cloudfoundry.credhub.audit.OperationDeviceAction;
import org.cloudfoundry.credhub.audit.RequestDetails;

public class DeleteCredential implements RequestDetails {
  private String name;

  public DeleteCredential() {
    super();

  }

  public DeleteCredential(final String name) {
    super();
    this.name = name;
  }

  public String getName() {
    return name;
  }

  public void setName(final String name) {
    this.name = name;
  }

  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.DELETE;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/DeleteCredential.kt
package org.cloudfoundry.credhub.audit.entities

import org.cloudfoundry.credhub.audit.OperationDeviceAction
import org.cloudfoundry.credhub.audit.RequestDetails

class DeleteCredential : RequestDetails {
    var name: String? = null

    constructor() : super() {
    }

    constructor(name: String) : super() {
        this.name = name
    }

    override fun operation(): OperationDeviceAction {
        return OperationDeviceAction.DELETE
    }
}
